### PR TITLE
Release 1.0.1 - Make makePersistable promisable

### PR DESCRIPTION
**Pull request by codeBelt**
@Zr9, @rafaeldsousa, @quarrant 

How does this look? If you can pull it down and test it out that would be helpful to double check that it works.

### CODE REVIEW

**Comment by rafaeldsousa** [comment]
@codeBelt Thanks @codeBelt, testing this now.

**Comment by rafaeldsousa** [comment]
@codeBelt It seems to work, but I'm getting a few new issues now.

1. Not being able to use `stringify: false` on `makePersistable`. LocalForage tries to clone the data object, but it fails saying 
`'IDBObjectStore': [object Object] could not be cloned.`. Looking further into the error, found [this](https://github.com/localForage/localForage/issues/610#issuecomment-252874521) and I indeed can see that when trying to store the data object it's referring to some data as Proxy data. This isn't happening on v1.0.0, not sure how this PR changes affects that.

2. I'm not being able to use a getter in the store to check if it has been hydrated (using the isHydrated method from the lib instead of setting a new observable). 
```js
get isHydrated() {
     return isHydrated(this);
}
 ```

File: src/PersistStore.ts [hunk 1]
```diff
@@ -67,11 +67,9 @@ export class PersistStore<T, P extends keyof T> {
...
       reactionOptions: this.reactionOptions,
     });
-
-    this.init();
   }
 
-  private async init() {
+  public async init(): Promise<void> {
     await this.hydrateStore();
     this.startPersisting();
```
Comment (quarrant) [inline src/PersistStore.ts:74]: I think we can return `this` there and use it later

File: src/makePersistable.ts [hunk 2]
```diff
@@ -18,5 +18,7 @@ export const makePersistable = <T extends { [key: string]: any }, P extends keyo
 
   PersistStoreMap.set(target, mobxPersistStore);
 
+  await mobxPersistStore.init();
```
Comment (quarrant) [inline src/makePersistable.ts:21]: You don't need call await and return mobxPersistStore on next line. After change init body we can return promise with PersistStore instance. 

```ts
return mobxPersistStore.init()
```

**Comment by zr9** [comment]
Hi sorry for delay, was quite busy with other stuff, but there is already pull req from you :)
Great job.

**Review by quarrant — APPROVED** [review]

**Pull request merged**